### ISSUE TITLE
Entering some characters in the search field crashes the React app and Heroic goes blank

### ISSUE THREAD

**Issue by arielj** [Heroic-Games-Launcher/HeroicGamesLauncher#1640]
### Describe the bug

Some characters like `[`, `]` and `\` make the React app crash.

Something seems to be generating a regexp with the input (probably the fuzzy search) and characters like those are not escaped, breaking the regexp parsing.

### Add logs

```
bundle.js:2987 Uncaught SyntaxError: Invalid regular expression: /[/: Unterminated character class (at bundle.js:2987:37)
    at new RegExp (<anonymous>)
    at bundle.js:2987:37
    at Array.filter (<anonymous>)
    at bundle.js:2987:25
    at updateMemo (bundle.js:88458:23)
    at Object.useMemo (bundle.js:88998:20)
    at useMemo (bundle.js:106714:25)
    at SearchBar (bundle.js:2985:62)
    at renderWithHooks (bundle.js:87599:22)
    at updateFunctionComponent (bundle.js:89921:24)
```

### Steps to reproduce

1. Focus the search input in the library
2. Type any character related to a regexp like `[` `]` `(` `)` `\` `*` `{` `}`

### Expected behavior

Search should work, maybe not showing results if nothing matches, but without crashing Heroic.

### System Information

- OS [e. g. "Ubuntu"]: Linux Mint
- Heroic Version [e. g. 2.1.1]: Latest main branch

**Comment by flavioislima** [comment]
This was fixed a long time ago but was probably reintroduced due to the fuzzy search like you said. I will investigate.

**Comment by flavioislima** [comment]
Fixed it on my PR #1635

**Comment by arielj** [comment]
Closing since this is fixed/merge in beta.
